pkg/restapi/vc/operation/operations.go

```diff
@@ -111,11 +160,71 @@ func createCredential(data *CreateCrendential) (*verifiable.Credential, error) {
 	return validatedCred, nil
 }
 
+func (c *Operation) createProfile(pr *ProfileRequest) (*ProfileResponse, error) {
+	if pr.DID == "" && pr.URI == "" {
+		return nil, fmt.Errorf("missing DID and URI information")
+	}
+
+	u, err := parseAndGetURI(pr.URI)
+	if err != nil {
+		return nil, err
+	}
+
+	issueDate := getIssueDate()
+	profileResponse := &ProfileResponse{
+		ID:        uuid.New().String(),
+		URI:       u,
+		IssueDate: &issueDate,
+		DID:       pr.DID,
+	}
+
+	err = c.profileStore.SaveProfile(profileResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	return profileResponse, nil
+}
+func parseAndGetURI(uri string) (string, error) {
+	u, err := url.Parse(uri)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse the uri: %s", err.Error())
+	}
+
+	u.Path = path.Join(u.Path, uuid.New().String())
+
+	return u.String(), nil
+}
+
+func getIssueDate() time.Time {
+	issueDate := time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(),
```

Comment:
no need to call time.Now() this many times
also, can you use time.Now().UTC()

Reply:
i have tried  time.Now().UTC() and it worked. I remember it didn't work before, so i have updated now.